### PR TITLE
Circle CI: Introduce default evm version parameter and change the default to `london`

### PR DESCRIPTION
**Pull request by bshastry**
; remove hard-coded setting.

### CODE REVIEW

**Review by cameel — DISMISSED** [review]
Looks good.

@chriseth suggested also changing `soltest.sh` so that we can just use `latest` instead of a specific version. This would save us having to update it. Might be done in a separate PR though.

File: .circleci/config.yml [hunk 1]
```diff
@@ -824,6 +829,8 @@ jobs:
       - store_artifacts: *artifacts_test_results
 
   t_ubu_ubsan_clang_cli:
+    environment:
+      UBSAN_OPTIONS: silence_unsigned_overflow=1
```
Comment (cameel) [inline .circleci/config.yml]: This seems unrelated to the PR.

Comment (bshastry) [reply .circleci/config.yml]: Yeah. I can remove it but felt it would be another PR just for this change. The context is that the nightly UBSan CLI job failed because UBSan printed superfluous unsigned integer overflows in boost onto stdout.

Comment (cameel) [reply .circleci/config.yml]: But the PRs would be independent, wouldn't they? One would not have to wait for the other.

I really think that it's better to have it in its own PR because if I wanted to get back to it, I would not even think about looking in this PR. Or at least to adjust the PR name to mention it.

Comment (bshastry) [reply .circleci/config.yml]: Sounds good, I will create a new PR for it.

**Review by cameel — APPROVED** [review]
